stefaniaPopescu26/assignment2/Ancestors.java

```diff
@@ -0,0 +1,63 @@
+import java.util.Scanner;
+import java.util.Vector;
+
+/*
+    When there are no ancestors to print, the printAncestors method
+    will print "No ancestors to show": for the root node or
+    for a node that doesn't exist.
+ */
+
+public class Ancestors<T> extends Tree<T>{
+
+    public Vector<Node<T>> getAncestors(T data) {
+        Vector<Node<T>> ancestors = new Vector<>();
+        Node<T> node = findNode(getRoot(), data);
+
+        if (node == null) {
```

Comment:
As I already wrote in the previous comment: this will make calling getAncestors with a data that is not even in the tree return an empty list. How is that distinguishable from calling getAncestors on the root Node? I think it would be good if the algorithm behaved differently for those two cases.